packages/core/src/ExpansionHub.ts

```diff
@@ -322,26 +324,62 @@ export interface ExpansionHub extends RevHub {
     getMotorEncoderPosition(motorChannel: number): Promise<number>;
 
     /**
-     * Set the current PID coefficients for a motor
+     * Set the Closed Loop Control Coefficients for Legacy PID mode.
+     *
      * @param motorChannel
      * @param motorMode
-     * @param pid
+     * @param algorithm Legacy PID algorithm
+     * @param pid Legacy PID coefficients
      */
-    setMotorPIDCoefficients(
+    setMotorClosedLoopControlCoefficients(
         motorChannel: number,
-        motorMode: MotorMode,
+        motorMode: number,
+        algorithm: ClosedLoopControlAlgorithm.LegacyPid,
         pid: PidCoefficients,
     ): Promise<void>;
 
     /**
-     * Get the current PID coefficients for a motor
+     * Set the Closed Loop Control Coefficients for PIDF mode
+     *
      * @param motorChannel
      * @param motorMode
+     * @param algorithm PIDF algorithm
+     * @param pidf PIDF coefficients
      */
-    getMotorPIDCoefficients(
+    setMotorClosedLoopControlCoefficients(
         motorChannel: number,
-        motorMode: MotorMode,
-    ): Promise<PidCoefficients>;
+        motorMode: number,
+        algorithm: ClosedLoopControlAlgorithm.Pidf,
+        pidf: PidfCoefficients,
+    ): Promise<void>;
+
+    /**
+     * Set Motor Closed Loop Coefficients using either Legacy PID algorithm or
+     * PIDF algorithm.
+     *
+     * @param motorChannel
+     * @param motorMode
+     * @param algorithm either Legacy PID or PIDF algorithm.
+     * @param pidf either PidCoefficients or PidfCoefficients depending on algorithm.
+     */
+    setMotorClosedLoopControlCoefficients(
+        motorChannel: number,
+        motorMode: number,
+        algorithm: ClosedLoopControlAlgorithm,
+        pidf: any,
+    ): Promise<void>;
+
+    /**
+     * Get motor closed loop control coefficients.
+     * Will return either PidCoefficients or PidfCoefficients depending on which
+     * algorithm is active.
+     * @param motorChannel
+     * @param motorMode
+     */
+    getMotorClosedLoopControlCoefficients(
+        motorChannel: number,
+        motorMode: number,
+    ): Promise<PidfCoefficients | PidCoefficients>;
```

Comment:
Should explictily return an algorithm alongside the parameters.

Reply:
the coefficients classes have an algorithm property now.